Formula/atop.rb

```diff
@@ -0,0 +1,48 @@
+class Atop < Formula
+  desc "Advanced system and process monitor for Linux using process events"
+  homepage "https://www.atoptool.nl"
+  url "https://github.com/Atoptool/atop/archive/refs/tags/v2.6.0.tar.gz"
+  sha256 "134de16cd58fb651d952e8ac401b20fe7f724afc0b152f06486e5c6f7be9c31a"
+  license "GPL-2.0-or-later"
+  head "https://github.com/Atoptool/atop.git", branch: "master"
+
+  depends_on :linux
+  depends_on "ncurses"
+  depends_on "zlib"
+
+  def install
+    if build.head?
+      inreplace "version.h" do |s|
+        s.sub!(/"$/, "-#{Utils.git_short_head}\"")
+      end
+    end
+    # Some used paths are hard-coded in several places, we have to replace them:
+    inreplace Dir["Makefile", "*.*"] do |s|
+      s.gsub!(%r{/var}, var.to_s, false)
+      s.gsub!(%r{/etc/default}, etc.to_s, false)
+    end
+    # The Makefile tries to run systemctl, but brew install blocks access to systemd.
+    # We have to fake successful calls:
+    inreplace "Makefile", "systemctl", "true"
+    inreplace "Makefile", "chown root", "true"
+    # Users have to install the service to e.g. ~/.config/systemd/user/ and enable it
+    # if they want to record system activity reports for atopsar report processing:
+    inreplace "atop.service", "/usr/bin/atop", "#{bin}/atop"
+    ENV["BINPATH"] = bin.to_s
+    ENV["SBINPATH"] = bin.to_s
+    ENV["MAN1PATH"] = man1.to_s
+    ENV["MAN5PATH"] = man5.to_s
+    ENV["MAN8PATH"] = man8.to_s
+    ENV["SYSDPATH"] = etc.to_s
+    ENV["PMPATHD"] = "prev"
+    system "make", "-e", "systemdinstall"
```

Comment:
Can't we just have a `service do` block that does the same as the systemd unit?

Reply:
As far as I have read, homebrew services only support plist, which is macos-only, and not systemd or sysvinit, depending on the init system. 
It seems there was work towards this direction but it does not seem to be integrated:
https://github.com/gergelyrozsas/homebrew-systemctl
At least the doc I found said that only plist on macos is supported.

Reply:
Also, if a normal user wants to install a service like this which runs as a daemon independent of user processes (and should be restart when the system boots), the only option is to use a systemd user unit. I know no other facility which can do that on Linux (atop is Linux-only).

In any case, the background recording for atopsar is not a core feature and everything needed to install the serivce is provided by this Formula on brew install atop.

Reply:
I haven't heard any complaints after https://github.com/Homebrew/homebrew-services/pull/350 so I guess it also works well on Linux (as a wrapper around SystemD).

Reply:
I've ran that service now for a day, and I must say that it creates large files which would fill up the user's home over time.
To have that properly rotated, additional services need to be installed, and this is too much effort to get it right for now.

So, at the moment, so service is automatically enabled, and I won't commit enabling it automatically.

atop is a tool like top and htop, just for different needs and atopsar can be enabled by experts which need it.

For me, I won't enable that atop service, it's too dangerous and nobody assumes to get a services which generates tons of data in one's home when just installing atop. Thus I won't do more.

I am open to remove the service files completely, but those who want to try the service may find it helpful.

I could install them as examples to a documentation of the package if you want, but the current location in etc is ok too.

Reply:
> I've ran that service now for a day, and I must say that it creates large files which would fill up the user's home over time.

So could nginx if configured incorrectly, but we still make a service available for it. The `service` block would just ensure that users can use the service, it doesn't enable it for them by default.

Reply:
Somebody else who has the motivation for that can do that. Maybe you want to transcribe this into a do service:

https://github.com/Atoptool/atop/blob/master/atop.service

Reply:
This systemd *timer* would be needed as well. How would you convert that to a `do service` and back to a systemd service?

https://github.com/Atoptool/atop/blob/master/atop-rotate.timer
```
[Unit]
Description=Daily atop restart

[Timer]
OnCalendar=daily

[Install]
WantedBy=timers.target
```

And this as well: https://github.com/Atoptool/atop/blob/master/atop-rotate.service

Really, I think trying to put the richness of systemd timers, units (eg. socket activation) and timer into a do service is just noy really working for everything. It would have been better to just have support for converting system-wide systemd service configurations into user configurations.